tracer/src/Datadog.Tracer.Native/clr_helpers.cpp

```diff
@@ -1004,4 +1004,41 @@ HRESULT FunctionLocalSignature::TryParse(PCCOR_SIGNATURE pbBase, unsigned len, s
     return S_OK;
 }
 
+HRESULT HasAsyncStateMachineAttribute(const ComPtr<IMetaDataImport2>& metadataImport, const mdMethodDef methodDefToken)
+{
+    const void* ppData = nullptr;
+    ULONG pcbData = 0;
+    const auto hasAsyncStateMachineAttrHr = metadataImport->GetCustomAttributeByName(
+        methodDefToken, WStr("System.Runtime.CompilerServices.AsyncStateMachineAttribute"), &ppData, &pcbData);
+
+    if (hasAsyncStateMachineAttrHr == S_OK)
+    {
+        if (pcbData > 0)
+        {
+            return S_OK;
+        }
+        return S_FALSE;
+    }
+
+    return hasAsyncStateMachineAttrHr;
+}
+
+HRESULT IsCompilerGeneratedType(const ComPtr<IMetaDataImport2>& metadataImport, const mdTypeDef typeDefToken)
```

Comment:
Why can't this method just return bool? We aren't exposing it as part of a COM interface, are we?

Reply:
Method not in use. Deleted in 5a103fbe57cedbc9e13568d4372c82a921c46e26